.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,82 @@
+name: Unit Tests
+on:
+  push:
+    branches:
+      # A test branch for seeing if your tests will pass in your personal fork
+      - test_me_github
+  pull_request:
+    branches:
+      - main
+      - master
+jobs:
+  docker-rspec:
+    runs-on:
+      - ubuntu-18.04
+    strategy:
+      matrix:
+        ruby:
+          - 2.7
+          - 2.6
+          - 2.5
+          - 2.4
+        docker_version:
+          - ':20.'
+          - ':19.'
+          - ':18.'
+      fail-fast: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - name: install bundler
+        run: |
+          gem install bundler -v '~> 1.17.3'
+          bundle update
+      - name: install docker
+        env:
+          DOCKER_VERSION: ${{ matrix.docker_version }}
+        run: |
+          set -x
+          sudo apt-get remove -y docker docker-engine docker.io containerd runc ||:
+          sudo apt-get update -y
+          sudo apt-get install -y apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+          sudo apt-get update -y
+          sudo apt-cache gencaches
+          sudo apt-get install -y docker-ce=$( apt-cache madison docker-ce | grep -e $DOCKER_VERSION | cut -f 2 -d '|' | head -1 | sed 's/\s//g' )
```

Comment:
Maybe we can also grep here based on `lsb_release`? Are we expecting it to be `bionic` or `focal`? I assume `bionic` because of the `runs-on` for the `docker-rspec` section.

Reply:
The repo setup on Line 45 targets it for whatever it's running. Installation outside of that  won't work.

`ubuntu-latest` seemed to give me varying results so I just gave up and queried whatever I had.

Reply:
@tlunter I think this should be OK if the message from yesterday made sense :point_up:

Reply:
Oh yes, definitely. Didn't see this was after the change that made the CI tests pass again. Thanks for resolving.